### PR TITLE
release: add 1.5.0 release notes

### PR DESCRIPTION
**Pull request by miekg**
Add the release notes to the notes/ directory to prepare for a release.

This is copied from the Wiki

### CODE REVIEW

File: notes/coredns-1.5.0.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+A new [release](https://github.com/coredns/coredns/releases/tag/v1.4.0): CoreDNS-1.5.0.
+
+**Two** new plugins in this releasse: [*grpc*](/plugins/grpc), and [*ready*](/plugins/ready). And
+some polish and simplications in the core server code.
+
+# Plugins
+
+* a new [*ready*](/plugins/ready) was added that signals a plugin is ready to receive queries. First user is the *kubernetes* plugin.
+* a new [*grpc*](/plugins/grpc) was added to implements forwarding gRPC. If you were relying on this in the [*proxy*](/explugins/proxy) plugin you can now use this one.
```
Comment (stp-ip) [inline notes/coredns-1.5.0.md]:
```suggestion
* a new [*grpc*](/plugins/grpc) was added to implement forwarding gRPC. If you were relying on this in the [*proxy*](/explugins/proxy) you can migrate to this one.
```

**Review by stp-ip — APPROVED** [review]
/lgtm

File: notes/coredns-1.5.0.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+date = "2019-03-31T09:39:07+00:00"
+author = "coredns"
++++
+
+A new [release](https://github.com/coredns/coredns/releases/tag/v1.4.0): CoreDNS-1.5.0.
+
+**Two** new plugins in this releasse: [*grpc*](/plugins/grpc), and [*ready*](/plugins/ready). And
+some polish and simplications in the core server code.
+
+# Plugins
```
Comment (ihac) [inline notes/coredns-1.5.0.md:18]: Seems we should announce the final deprecation of `TIMEOUT` and `NO_RELOAD` in this release note.

**Comment by fturib** [comment]
The same way as `timeout` and `no_reload`,
I guess we should also mentioned that `proxy` is no more included in the image, and using the option `upstream` will not be recognized.

File: notes/coredns-1.5.0.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+description = "CoreDNS-1.5.0 Release Notes."
+tags = ["Release", "1.5.0", "Notes"]
+release = "1.5.0"
+date = "2019-03-31T09:39:07+00:00"
+author = "coredns"
++++
+
+A new [release](https://github.com/coredns/coredns/releases/tag/v1.5.0): CoreDNS-1.5.0.
+
+**Two** new plugins in this releasse: [*grpc*](/plugins/grpc), and [*ready*](/plugins/ready). And
```
Comment (chrisohaver) [inline notes/coredns-1.5.0.md]:
```suggestion
**Two** new plugins in this release: [*grpc*](/plugins/grpc), and [*ready*](/plugins/ready). And
```

**Comment by fturib** [comment]
About deprecation of `no_reload`, `timeout` and `proxy`, 
Are we following our own [deprecation policy]
(https://github.com/coredns/coredns/blob/master/README.md#deprecation-policy) ?

I understand from the document that for CoreDNS 1.5.0, the feature would be deactivated, but the corefile should still be compatible:
> Release x.y+1.0: Increase the minor version and set the patch version to 0. Make the changes, but allow the old configuration to be parsed. I.e. CoreDNS will start from an unchanged Corefile.


hum... looks like already [discussed here ](https://github.com/coredns/coredns/pull/2742#issuecomment-478348891) and decided to go ahead.

**Comment by miekg** [comment]
Proxy was long gone

**Comment by chrisohaver** [comment]
Proxy was deprecated in 1.4.0... In what release is it removed?

What does "fully deprecated" mean?  Removed?  I thought deprecated generally means that the feature is still present, but its use is discouraged.

**Pull request merged**